Public/Src/Cache/ContentStore/Library/Utils/ResultsExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using BuildXL.Cache.ContentStore.Interfaces.Results;
+using BuildXL.Cache.ContentStore.Synchronization;
+
+namespace BuildXL.Cache.ContentStore.Utils
+{
+    internal static class ResultsExtensions
+    {
+        /// <nodoc />
+        public static TResult WithLockAcquisitionDuration<TResult, TLockKey>(this TResult result, in LockSet<TLockKey>.LockHandle handle)
+            where TResult : ResultBase
+            where TLockKey : IEquatable<TLockKey>
+        {
+            if (handle.LockAcquisitionDuration != null)
+            {
+                var message = $", lock acquired by {(long)handle.LockAcquisitionDuration.Value.TotalMilliseconds}ms, ";
+                result.ExtraDiagnosticsMessage = message;
+            }
+
+            return result;
+        }
+
+        /// <nodoc />
+        public static string GetExtraDiagnosticsMessageForTracing(this ResultBase result)
+        {
+            if (!string.IsNullOrEmpty(result.ExtraDiagnosticsMessage))
+            {
+                return " " + result.ExtraDiagnosticsMessage;

```

Comment:
>return " " + result.ExtraDiagnosticsMessage; [](start = 16, length = 44)

Is it true that an example message would read:
`Blah.OpenStream stop 42ms , lock acquired by 2ms,  result=[Success]` ?

Can we pare this down? Perhaps remove the two commas, the extra space, and " acquired by".

Reply:
Yep, I've messed up. The output right now is `104ms , lock acquired by 0ms,  result=`.  I'll change it.